### DISCUSSION TITLE
bounding box center

### DISCUSSION THREAD

**Discussion by pca006132** [elalish/manifold#538] [Q&A] [answered]
If I understand it correctly, the collider expects the leaf nodes to be ordered by the morton code of the box center, which can be different from the center of a triangle.

https://github.com/elalish/manifold/blob/1514c151840356398a08c798fd630caeaea38c7f/src/manifold/src/sort.cpp#L84-L113

@elalish am I misunderstanding something here or is this a bug?

**Comment by elalish** [comment, marked as answer]
It actually doesn't matter - the bounding boxes are what actually get queried against, and so long as they are unioned properly up the tree, the result will be correct. The Morton code is about sorting such that those unioned bounding boxes tend to stay small, thus keeping the BVH efficient. Using centers does a decent job, and there's way more variance introduced by the z-order curve, so in practice it doesn't matter much.

**Reply by pca006132** [reply]
oh ok, because I am trying to implement something related to finding minimum gap, so I have to use the BVH stuff and was worried that using an incorrect center will cause the BVH to break.
